src/EcommerceStatsBundle/DataModel/CsvResponse.php

```diff
@@ -0,0 +1,58 @@
+<?php declare(strict_types=1);
+
+namespace ChameleonSystem\EcommerceStatsBundle\DataModel;
+
+use Symfony\Component\HttpFoundation\Response;
+use Symfony\Component\HttpFoundation\ResponseHeaderBag;
+
+class CsvResponse extends Response
+{
+
+    public static function fromRows(string $fileName, array $data, string $separator = ';'): CsvResponse
+    {
+        $csv = fopen('php://temp/maxmemory:'. 1024 * 1024, 'r+');
```

Comment:
Is this faster (less memory) than simply collecting it in an array (and merge at the end)?

Reply:
At some point we have to take the code and CSV encode it, which is here. Going the temporary file route allows us to use `fputcsv` which in turn takes care of escaping and quoting of the data rows. 

While symfony has a very handy `CsvEncoder` in it's serializer component, we cannot use it here, since that component assumes that a header line is always added to the content, which is not the case here.